sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/UnsafeRow.java

```diff
@@ -363,15 +363,19 @@ public Decimal getDecimal(int ordinal, int precision, int scale) {
   @Override
   public UTF8String getUTF8String(int ordinal) {
     assertIndexIsValid(ordinal);
-    return isNullAt(ordinal) ? null : UTF8String.fromBytes(getBinary(ordinal));
```

Comment:
Is there any chance that avoiding this copy could lead to a bug?  It's probably unlikely, given that we'd probably use separate UnsafeRow pointers when comparing two rows.